src/main/java/unibook/logic/commands/EditCommand.java

```diff
@@ -57,6 +89,69 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editModuleDescriptor = null;
+    }
+
+    /**
+     * @param index of the person in the filtered module list to edit
+     * @param editModuleDescriptor details to edit the module with
+     */
+    public EditCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = index;
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+        this.editPersonDescriptor = null;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException, ModuleNotFoundException {
+        requireNonNull(model);
+
+        if (this.editModuleDescriptor == null) {
+            List<Person> lastShownList = model.getFilteredPersonList();
+            List<Module> latestModList = model.getFilteredModuleList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            Person personToEdit = lastShownList.get(index.getZeroBased());
+            if (!editPersonDescriptor.getModules().equals(Optional.empty()) && latestModList.size() != 0) {
+                Module checkMod = editPersonDescriptor.getModules().get().iterator().next();
+                if (!latestModList.contains(checkMod)) {
+                    throw new ModuleNotFoundException();
+                }
+            }
+            Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+
+            if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+                throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            }
+
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+
+        } else {
+            List<Module> lastShownList = model.getFilteredModuleList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
+            }
+
+            Module moduleToEdit = lastShownList.get(index.getZeroBased());
+            Module editedModule = createEditedModule(moduleToEdit, editModuleDescriptor);
+
+            if (!moduleToEdit.isSameModule(editedModule) && model.hasModule(editedModule)) {
+                throw new CommandException(MESSAGE_DUPLICATE_MODULE);
+            }
+
+            model.setModule(moduleToEdit, editedModule);
+            model.updateFilteredModuleList(PREDICATE_SHOW_ALL_MODULES);
```

Comment:
As mentioned in previous comment, if the module object is changed due to being edited, then all the associated modules to each person object will now be pointing to an outdated module object. This means the change will not be reflected in their lists of associated modules, and references to the old module object will still exist, not getting garbage collected.

You need to go through each person in the module, and replace their old module object in their list of associated modules with the new module object you just created here